db_stress_tool/expected_value.cc

```diff
@@ -10,17 +10,32 @@
 #include <atomic>
 
 namespace ROCKSDB_NAMESPACE {
-void ExpectedValue::Put(bool pending) {
+bool ExpectedValue::Put(bool pending) {
+  // Last write or delete didn't finish due to errors (e.g, IO error)
+  // TODO (hx235): if the error is recoverable, we should close the pendig state
+  // upon error recovery finishes
+  if (pending && (PendingWrite() || PendingDelete())) {
+    return false;
```

Comment:
This is to simplify the problem for now. The pending state will be close in VerifyOrSyncValue() as if we crashed during the pending state.